src/main/scala/rx/lang/scala/Emitter.scala

```diff
@@ -0,0 +1,44 @@
+package rx.lang.scala
+
+import rx.annotations.Experimental
+import rx.functions.Cancellable
+
+@Experimental
+trait Emitter[-T] extends Observer[T] {
```

Comment:
Now that I think about it, not sure this interface carries it's weight for a scala version.

It's nearly identical, but differs in 3 ways:

setCancellation takes a () => Unit instead of a rx.functions.Cancellable SAM type. This is super minor for 2.12 users, as Scala can automatically convert the function type to the SAM type.

setSubscription takes a rx.lang.scala.Subscription instead of a rx.Subscription. This is annoying for users, as they would need to import JavaConvertors and always use subscription.asJava. However, it is not very difficult and setSubscription is a tad more advanced usage which people won't always use.

It inherits from rx.lang.scala.Observer instead of rx.Observer. I have now idea how common/valid the use case of passing it around as an Observer is , but I suspect usually people will only use the interface directly in Observable.fromEmitter(e => /* use e */) so it isn't a problem